### ISSUE TITLE
differ between integer & number

### ISSUE THREAD

**Issue by FlashHit** [Imposter/vscode-lua-vu#13]
atm every `int` & `float` gets the type `---@type number`.
It would be better to give `int` the type `---@type integer` and `float` the type `---@type number`. That way it's easier to know what vext is expecting.

**Comment by FlashHit** [comment]
so instead of
```
---@class SoldierWeaponsComponentData:ComponentData
---@field aimDir LinearTransform
---@field weaponSkeleton SkeletonAsset|nil
---@field primaryWeaponId number
---@field animatedWeaponBinding AnimatedWeaponBinding
---@field animated1pOnlyWeaponBinding Animated1pOnlyWeaponBinding
---@field animated3pOnlyWeaponBinding Animated3pOnlyWeaponBinding
---@field lockTimeMultiplier number
---@field grenadeIncrease number
---@field ammoClipIncreaseMultiplier number
---@field explosiveIncreaseMultiplier number
---@field underslungGrenadeIncrease number
---@field useExternalAimDir boolean
---@field unlimitedAmmo boolean
---@field unlimitedMags boolean
---SoldierWeaponsComponentData (Class)
SoldierWeaponsComponentData = {}
```

it should be 
```
---@class SoldierWeaponsComponentData:ComponentData
---@field aimDir LinearTransform
---@field weaponSkeleton SkeletonAsset|nil
---@field primaryWeaponId integer
---@field animatedWeaponBinding AnimatedWeaponBinding
---@field animated1pOnlyWeaponBinding Animated1pOnlyWeaponBinding
---@field animated3pOnlyWeaponBinding Animated3pOnlyWeaponBinding
---@field lockTimeMultiplier number
---@field grenadeIncrease integer
---@field ammoClipIncreaseMultiplier number
---@field explosiveIncreaseMultiplier number
---@field underslungGrenadeIncrease integer
---@field useExternalAimDir boolean
---@field unlimitedAmmo boolean
---@field unlimitedMags boolean
---SoldierWeaponsComponentData (Class)
SoldierWeaponsComponentData = {}
```

**Comment by Imposter** [comment]
Fixed in 0bc6068
